### PR TITLE
OCPBUGS-37703: fix release stream flag/rhcos image for create cluste…

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**
**What this PR does / why we need it**:  individually looks up OCP version based on the release stream so that RHCOS image isn't empty when creating an azure cluster.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes # OCPBUGS-37703](https://issues.redhat.com/browse/OCPBUGS-37703)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: cmd/cluster/azure/create.go [hunk 1]
```diff
@@ -371,7 +372,15 @@ func CreateInfraOptions(ctx context.Context, azureOpts *ValidatedCreateOptions,
 }
 
 // lookupRHCOSImage looks up a release image and extracts the RHCOS VHD image based on the nodepool arch
-func lookupRHCOSImage(ctx context.Context, arch string, image string, pullSecretFile string) (string, error) {
+func lookupRHCOSImage(ctx context.Context, arch, image, releaseStream, pullSecretFile string) (string, error) {
```
Comment (jparrill) [inline cmd/cluster/azure/create.go:375]: It could be possible to cover the function with a unit test?

Comment (Patryk-Stefanski) [reply cmd/cluster/azure/create.go:375]: This fix is blocking our azure e2e so I would prefer to add a unit test in a separate pr later on.

File: cmd/cluster/azure/create.go [hunk 2]
```diff
@@ -371,7 +372,15 @@ func CreateInfraOptions(ctx context.Context, azureOpts *ValidatedCreateOptions,
...
+	if len(image) == 0 && len(releaseStream) != 0 {
+		defaultVersion, err := version.LookupDefaultOCPVersion(releaseStream)
+		if err != nil {
+			return "", fmt.Errorf("release image is required when unable to lookup default OCP version: %w", err)
```
Comment (bryan-cox) [inline cmd/cluster/azure/create.go:379]: Should this say either a release image or a release stream should be provided?

Comment (bryan-cox) [inline cmd/cluster/azure/create.go:379]:
```suggestion
			return "", fmt.Errorf("failed to lookup OCP image for release stream, %s: %w", releaseStream, err)
```

File: cmd/cluster/azure/create.go (hunk 1 above)
Comment (sjenning) [reply cmd/cluster/azure/create.go:375]: This is done in the core create
https://github.com/openshift/hypershift/blob/cbf39ff40a564fc4a99dfd7b0c4e1b335538d39d/cmd/cluster/core/create.go#L210-L216

It seems that Azure needs this at the infra create stage, which is why we need to do it here?

**Comment by sjenning** [comment]
/test unit

`TestValidateKeyPairItempotency` flake

**Pull request merged**